libvips/colour/profile_load.c

```diff
@@ -240,8 +240,8 @@ vips__profile_set( VipsImage *image, const char *name )
 	if( profile ) { 
 		data = ((VipsArea *) profile)->data;
 		length = ((VipsArea *) profile)->length;
-		vips_image_set_blob( image, VIPS_META_ICC_NAME, 
-			(VipsCallbackFn) NULL, data, length );
+		vips_image_set_blob_copy( image, VIPS_META_ICC_NAME,
+			data, length );
```

Comment:
Argh you're right, this is wrong too, but it will cause a copy. Really we want to just up the refcount on the blob.

How about:

```C
        GValue value = { 0 };

        g_value_init( &value, VIPS_TYPE_BLOB );
        g_value_set_boxed( value, profile );
        vips_image_set( image, VIPS_META_ICC_NAME, &value );
        g_value_unset( &value );
```

Reply:
Thanks John, I've updated the PR with this approach.